src/consensus/blockchain.py

```diff
@@ -501,3 +509,70 @@ async def validate_unfinished_block(
             return None, error_code
 
         return required_iters, None
+
+    def sub_block_record(self, header_hash: bytes32) -> SubBlockRecord:
+        sub_block = self.__sub_blocks[header_hash]
+        assert sub_block is not None
+        return sub_block
+
+    def height_to_sub_block_record(self, sub_height: uint32) -> SubBlockRecord:
+        header_hash = self.sub_height_to_hash(sub_height)
+        assert header_hash is not None
+        return self.sub_block_record(header_hash)
+
+    def get_ses_heights(self) -> List[uint32]:
+        return sorted(self.__sub_epoch_summaries.keys())
+
+    def get_ses(self, height: uint32) -> SubEpochSummary:
+        return self.__sub_epoch_summaries[height]
+
+    def get_ses_from_height(self, height: uint32) -> List[SubEpochSummary]:
+        ses_l = []
+        for ses_height in reversed(self.get_ses_heights()):
+            if ses_height <= height:
+                break
+            ses_l.append(self.get_ses(ses_height))
+        return ses_l
+
+    def sub_height_to_hash(self, height: uint32) -> Optional[bytes32]:
+        if height not in self.__sub_height_to_hash:
+            log.warning("could not find header hash in cache")
+            return None
+        return self.__sub_height_to_hash[height]
+
+    def contains_sub_height(self, height: uint32) -> bool:
+        return height in self.__sub_height_to_hash
+
+    def get_peak_height(self) -> Optional[uint32]:
+        return self.peak_height
+
+    async def forkpoint_warmup(self, fork_point: uint32):
+        # load all blocks such that fork - self.constants.SUB_BLOCKS_CACHE_SIZE -> fork in dict
+        blocks = await self.block_store.get_sub_block_in_range(
+            fork_point - self.constants.SUB_BLOCKS_CACHE_SIZE, fork_point
+        )
+        self.__sub_blocks.update(blocks)
```

Comment:
Update doesn't delete any sub blocks, so you will end up with a cache 2x as big. is this intentional?